packages/graphql-codegen-typescript-typemap-plugin/src/index.ts

```diff
@@ -17,7 +17,10 @@ const graphqlCodegenTypeMapPlugin: PluginFunction = (
           `  "${typeName}": ${
             isScalarType(typesMap[typeName])
               ? `Scalars["${typeName}"]`
-              : pascalCase(typeName)
+              : typeName
+                  .split("_")
+                  .map((part) => pascalCase(part))
+                  .join("_")
```

Comment:
I think we should probably be using `@graphql-codegen` utility functions for the naming, but it was hard to figure out so I hadn't put in more work at the time. This seems fine for now 👍

Reply:
Let me know exactly what you would want, maybe I will find time to do it properly. Which utility functions did have in mind? Any pointers?

Reply:
Their "base visitor" has a bunch of utilities, but it wasn't clear to figure out which ones to use and how. Here's the public config docs https://the-guild.dev/graphql/codegen/docs/config-reference/naming-convention